### PR TITLE
CSI-Powerflex installation using centralize helm

### PR DESCRIPTION
**Pull request by KshitijaKakde**
# Description
Installation using centralize helm

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/877

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
image
image
image
image

### CODE REVIEW

**Review by VamsiSiddu-7 — DISMISSED** [review]

**Review by ashleyvjoy — APPROVED** [review]

File: dell-csi-helm-installer/csi-install.sh [hunk 1]
```diff
@@ -399,8 +410,8 @@ helm --help >&/dev/null || {
 OPENSHIFT=$(isOpenShift)
 
 # Get the kubernetes major and minor version numbers.
-kMajorVersion=$(run_command kubectl version | grep 'Server Version' | sed -E 's/.*v([0-9]+)\.[0-9]+\.[0-9]+.*/\1/')
-kMinorVersion=$(run_command kubectl version | grep 'Server Version' |  sed -E 's/.*v[0-9]+\.([0-9]+)\.[0-9]+.*/\1/')
+kMajorVersion=$(run_command kubectl version | grep 'Server Version' | sed -e 's/^.*Major:"//' -e 's/[^0-9].*//g')
```
Comment (VamsiSiddu-7) [inline dell-csi-helm-installer/csi-install.sh:413]: Why are you changing this regex?

File: dell-csi-helm-installer/csi-install.sh [hunk 2]
```diff
@@ -13,13 +13,35 @@
...
 
 SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
-DRIVERDIR="${SCRIPTDIR}/../helm"
+DRIVERDIR="${SCRIPTDIR}/../"
+HELMCHARTAG="csi-vxflexos-2.8.0"
+
+if git ls-remote --quiet --tag  https://github.com/dell/helm-charts | grep -q "tags/${HELMCHARTAG}"; then
+  REPOREF="${HELMCHARTAG}"
+else
+  REPOREF="release-v1.8.0"
```
Comment (VamsiSiddu-7) [inline dell-csi-helm-installer/csi-install.sh:22]: release-v1.8.0 should not be hardcoded it should be passed from a cmd line parameter. based on the parameter it should be set

File: dell-csi-helm-installer/csi-install.sh [hunk 3]
```diff
@@ -13,13 +13,35 @@
 # limitations under the License.
 
 SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
-DRIVERDIR="${SCRIPTDIR}/../helm"
+DRIVERDIR="${SCRIPTDIR}/../"
+HELMCHARTAG="csi-vxflexos-2.8.0"
+
+if git ls-remote --quiet --tag  https://github.com/dell/helm-charts | grep -q "tags/${HELMCHARTAG}"; then
```
Comment (VamsiSiddu-7) [inline dell-csi-helm-installer/csi-install.sh:19]: why do we need this check?

File: dell-csi-helm-installer/csi-offline-bundle.sh [hunk 4]
```diff
@@ -222,11 +227,32 @@ set_mode() {
...
 # some directories
 SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 REPODIR="$( dirname "${SCRIPTDIR}" )"
-HELMDIR="${REPODIR}/helm"
+HELMCHARTAG="csi-vxflexos-2.8.0"
+
+if git ls-remote --quiet --tag  https://github.com/dell/helm-charts | grep -q "tags/${HELMCHARTAG}"; then
+  REPOREF="${HELMCHARTAG}"
+else
+  REPOREF="release-v1.8.0"
```
Comment (VamsiSiddu-7) [inline dell-csi-helm-installer/csi-offline-bundle.sh:240]: same comments here

**Review by VamsiSiddu-7 — CHANGES_REQUESTED** [review]
Needs changes. refer the comments

File: dell-csi-helm-installer/csi-install.sh (hunk 3 above)
Comment (VamsiSiddu-7) [reply dell-csi-helm-installer/csi-install.sh:19]: the latest version will be always present

**Review by VamsiSiddu-7 — DISMISSED** [review]
Changes looks good to me i hope you tested both offline and helm.

**Review by khareRajshree — APPROVED** [review]

**Pull request merged**